### ISSUE TITLE
Account creation via social media logins could be potentially improved

### ISSUE THREAD

**Issue by kristapsk** [stackernews/stacker.news#841]
Restarted Google Chrome (memory leaks, you know) and somehow forgot which social media account I used to log in to SN. Turns out it was Twitter. But I first tried GitHub. Without asking any questions it created new account kristapsk2 and now I can't even link GitHub to my original kristapsk account ("This auth method is linked to another account. To link to this account first unlink the other account"). Don't have concrete suggestions here, but UX in this regard could be improved.

**Comment by ekzyis** [comment]
Yes, you are not the first one who is confused by new accounts getting automatically created. Login and signup are currently doing basically the same. I think the solution would be to only create accounts when signup is used. If the auth method was already used, it should throw an error.[^1] Vice versa, when login is used, it should throw an error when the auth method is not used already.

> it created new account kristapsk2 and now I can't even link GitHub to my original kristapsk account

I think you need to login with your Github account again to access the wrong account and then unlink the auth method as mentioned in the error message with "first unlink the other account". Then you should be able to link Github to your actual account.

Did you try that?

[^1]: in a way that preserves privacy as much as possible but a leak like "account already exists" is probably not preventable to make this change worthwhile

**Comment by huumn** [comment]
@kristapsk I owe you some sats. Do you have a lightning address I can send them to?
